webrtc.html

```diff
@@ -6620,6 +6619,45 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
         </li>
       </ol>
 
+      <p>Additionally, when the <a>ICE Agent</a> indicates that it finished
+      gathering an individual generation of candidates for an
+      <code><a>RTCIceTransport</a></code>, the User Agent MUST queue
+      a task that runs the following steps:</p>
+      <ol>
+        <li>
+          <p>Let <var>connection</var> be the
+          <code><a>RTCPeerConnection</a></code> object associated with this
+          <a>ICE Agent</a>.</p>
+        </li>
+        <li>
+          <p>If <var>connection</var>'s [[<a>isClosed</a>]] slot is
+          <code>true</code>, abort these steps.</p>
+        </li>
+        <li>
+          <p>Let <var>transport</var> be the relevant
+          <code><a>RTCIceTransport</a></code>.</p>
+        </li>
+        <li>
+          <p>Create an <code><a>RTCIceCandidate</a></code> instance, with
+          <code><a data-for="RTCIceCandidate">sdpMid</a></code> and
+          <code><a data-for="RTCIceCandidate">sdpMLineIndex</a></code>
+          set to the values associated with this
+          <code><a>RTCIceTransport</a></code>, with
+          <code><a data-for="RTCIceCandidate">ufrag</a></code> set to the ufrag
+          of the generation of candidates for which gathering finished, and
+          with all other nullable members set to null.</p>
+        </li>
+        <li>
+          <p>Fire an event named <code><a>icecandidate</a></code> with
+          <var>newCandidate</var> at <var>connection</var>.</p>
+        </li>
+      </ol>
+      <div class="issue">
+        ICEbis doesn't define what a "generation of candidates" is, though this
+        terminology is used in places like [[TRICKLE-ICE]]. Once a definition is
+        added we should reference it here.
+      </div>
+
```

Comment:
This can be seen as triggered on transport.icegatheringstate change, right? I think we should describe it as such, rather than "when it's finished", since that decides which comes first - the event or the state change.

If we add a trigger on pc.icegatheringsummarystate that fires an empty event (no candidate at all), we don't even have to ask implementations to change (assuming they don't barf on the empty candidate).